### PR TITLE
cmd/tailscale,ipn: store disallowed TKA's in prefs, lock local-disable

### PR DESCRIPTION
**Pull request by twitchyliquid64**
Take 2 of https://github.com/tailscale/tailscale/pull/6546, builds on https://github.com/tailscale/tailscale/pull/6560

This time with (hopefully) more greenage! :grinning:

### CODE REVIEW

File: client/tailscale/localclient.go [hunk 1]
```diff
@@ -853,6 +853,21 @@ func (lc *LocalClient) NetworkLockLog(ctx context.Context, maxEntries int) ([]ip
 	return decodeJSON[[]ipnstate.NetworkLockUpdate](body)
 }
 
+// NetworkLockForceLocalDisable forcibly shuts down network lock on this node.
+func (lc *LocalClient) NetworkLockForceLocalDisable(ctx context.Context) error {
+	// This endpoint expects an empty JSON stanza as the payload.
+	var b bytes.Buffer
+	if err := json.NewEncoder(&b).Encode(struct{}{}); err != nil {
+		return err
+	}
```
Comment (bradfitz) [inline client/tailscale/localclient.go:862]: I suppose that's one way to encode `{}` :joy:

File: ipn/ipnlocal/network-lock.go [hunk 2]
```diff
@@ -495,6 +510,31 @@ func (b *LocalBackend) NetworkLockKeyTrustedForTest(keyID tkatype.KeyID) bool {
...
+	id1, id2 := b.tka.authority.StateIDs()
+	stateID := fmt.Sprintf("%d:%d", id1, id2)
+
+	newPrefs := b.pm.CurrentPrefs().AsStruct().Clone() // .Persist should always be initialized here.
+	newPrefs.Persist.DisallowedTKAStateIDs = append(newPrefs.Persist.DisallowedTKAStateIDs, stateID)
+	if err := b.pm.SetPrefs(newPrefs.View()); err != nil {
+		return fmt.Errorf("saving prefs: %w", err)
+	}
+
+	if err := os.RemoveAll(b.chonkPathLocked()); err != nil {
```
Comment (bradfitz) [inline ipn/ipnlocal/network-lock.go:531]: chonkPathLocked can never return "/" or "." I hope?

File: tka/tka.go [hunk 3]
```diff
@@ -714,3 +714,8 @@ func (a *Authority) Keys() []Key {
 	}
 	return out
 }
+
+// StateIDs returns the stateIDs for this tailnet key authority.
+func (a *Authority) StateIDs() (uint64, uint64) {
```
Comment (bradfitz) [inline tka/tka.go:719]: // TODO(tom): document all this 

:)

**Review by bradfitz — COMMENTED** [review]
I have no clue what's going on here TKA-wise but I like green.

File: client/tailscale/localclient.go (hunk 1 above)
Comment (twitchyliquid64) [reply client/tailscale/localclient.go:862]: I guess its a lil resource wasteful compared to just yeeting a string of braces, but the latter feels dirty, idk lmao

File: ipn/ipnlocal/network-lock.go (hunk 2 above)
Comment (twitchyliquid64) [reply ipn/ipnlocal/network-lock.go:531]: Yepp! `filepath.Join(b.TailscaleVarRoot(), "tka-profiles", string(b.pm.CurrentProfile().ID))`

And you can only get that far if you have TKA initialized, which transitively implies that `b.TailscaleVarRoot()` references an accessible path

File: tka/tka.go (hunk 3 above)
Comment (twitchyliquid64) [reply tka/tka.go]: I've added a bit more verbaige, but most of the docs for stateID's are on the `State` struct. Theres more context on stateIDs specifically in https://tailscale.slack.com/archives/C03LXV1NTB8/p1668552550674699

**Review by sailorfrag — APPROVED** [review]

**Pull request merged**